include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -669,6 +669,15 @@ def ObserveAnsatz : Pass<"observe-ansatz", "mlir::func::FuncOp"> {
   ];
 }
 
+def DeleteStates : Pass<"delete-states", "mlir::ModuleOp"> {
+  let summary =
+    "Remove unnecessary creation of new states from data";
+  let description = [{
+    - Replace state by its data if possible
+    - Add state deletion in before exit if not possible to remove a new state
```

Comment:
I think this could be expanded a bit to be more informative as to what the pass is doing. An example of the problem it is solving might be useful here.

Reply:
will do, thanks!